### PR TITLE
chore: add version detection to `create fuels`

### PR DESCRIPTION
**Pull request by petertonysmith94**
Fixing [e2e](https://github.com/FuelLabs/fuels-ts/actions/runs/9514897546/job/26247701045#step:7:70) test that is failing.

### CODE REVIEW

**Review by arboleya — COMMENTED** [review]
It seems like the e2es are [still failing](https://github.com/FuelLabs/fuels-ts/actions/runs/9530498220/job/26270493227?pr=2524#step:7:69).

File: packages/create-fuels/src/lib/getPackageVersion.test.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+/**
+ * @group node
+ */
+describe('getPackageVersion', () => {
+  beforeEach(() => {
+    vi.resetAllMocks();
+  });
+
```
Comment (arboleya) [inline packages/create-fuels/src/lib/getPackageVersion.test.ts:49]:
```suggestion

  const prVersion = '0.0.0-pr-2524-20240616051538';
    
```

File: packages/create-fuels/src/lib/getPackageVersion.test.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  it('should get the version from an inline create-fuels binary', () => {
+    const binDir = 'node_modules/create-fuels';
+    const expectedPackageJsonPath = `${binDir}/package.json`;
+    const argv = ['node', `${binDir}/create-fuels.js`];
+    const { readFileSync, existsSync } = mockDeps({
+      binDir,
+      createFuelsVersion: '0.0.0-pr-2524-20240616051538',
```
Comment (arboleya) [inline packages/create-fuels/src/lib/getPackageVersion.test.ts:56]:
```suggestion
      createFuelsVersion: prVersion,
```

File: packages/create-fuels/src/lib/getPackageVersion.test.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    });
+    existsSync.mockReturnValueOnce(true);
+
+    const version = getPackageVersion(argv);
+
+    expect(readFileSync).toBeCalledWith(expectedPackageJsonPath, 'utf8');
+    expect(version).toBe('0.0.0-pr-2524-20240616051538');
```
Comment (arboleya) [inline packages/create-fuels/src/lib/getPackageVersion.test.ts:63]:
```suggestion
    expect(version).toBe(prVersion);
```

File: packages/create-fuels/src/lib/getPackageVersion.test.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  });
+
+  it('should get the version from a .bin relative create-fuels binary', () => {
+    const binDir = 'node_modules/.bin';
+    const expectedPackageJsonPath = `node_modules/create-fuels/package.json`;
+    const argv = ['node', `${binDir}/create-fuels`];
+    const { readFileSync, existsSync } = mockDeps({
+      binDir,
+      createFuelsVersion: '0.0.0-pr-2524-20240616051538',
```
Comment (arboleya) [inline packages/create-fuels/src/lib/getPackageVersion.test.ts:72]:
```suggestion
      createFuelsVersion: prVersion,
```

File: packages/create-fuels/src/lib/getPackageVersion.test.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    });
+    existsSync.mockReturnValueOnce(false);
+    existsSync.mockReturnValueOnce(true);
+
+    const version = getPackageVersion(argv);
+
+    expect(readFileSync).toBeCalledWith(expectedPackageJsonPath, 'utf8');
+    expect(version).toBe('0.0.0-pr-2524-20240616051538');
```
Comment (arboleya) [inline packages/create-fuels/src/lib/getPackageVersion.test.ts:80]:
```suggestion
    expect(version).toBe(prVersion);
```

**Review by arboleya — COMMENTED** [review]
Nits.

File: packages/create-fuels/src/lib/getPackageVersion.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+      throw new FuelError(
+        FuelError.CODES.PARSE_FAILED,
+        `Unable to find package.json at ${inlinePackageJsonPath} or ${relativeBinPackageJsonPath}`
+      );
+    }
+
+    const packageJson = readFileSync(path, 'utf8');
+    version = JSON.parse(packageJson).version;
+  } catch (error) {
+    version = versions.FUELS;
```
Comment (arboleya) [inline packages/create-fuels/src/lib/getPackageVersion.ts:38]: Why can't we _always_ use the version from `versions.FUELS`?

File: packages/create-fuels/src/cli.ts [hunk 7]
```diff
@@ -158,10 +160,11 @@ export const runScaffoldCli = async ({
   // Note: `pnpm run xprebuild` -> rewritten to -> `pnpm run prebuild` (on prePublish script)
   const packageJsonPath = join(projectPath, 'package.json');
   const packageJsonContents = readFileSync(packageJsonPath, 'utf-8');
-  const newPackageJsonContents = packageJsonContents.replace(
-    `pnpm run prebuild`,
-    packageManager.run('prebuild')
-  );
+  const fuelsVersion = getPackageVersion(args);
```
Comment (arboleya) [inline packages/create-fuels/src/cli.ts:163]: Can we use the version directly from the `versions` package?

```suggestion
  const fuelsVersion = versions.FUELS;
```

**Review by arboleya — APPROVED** [review]
LGTM — Later on, I'd like to review the overall approach tho.

I feel I'm a bit out of context.

**Review by danielbate — APPROVED** [review]

**Pull request merged**